specification/schemaregistry/data-plane/Microsoft.EventHub/stable/2023-07/schemaregistry.json

```diff
@@ -0,0 +1,615 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Schema Registry",
+    "version": "2023-07",
```

Comment:
Are we releasing this feature as GA or as preview? I believe this is the first time this feature is being released, so shouldn't the API version be a preview (e.g. `2023-07-01-preview`)?

Reply:
Technically this feature is in preview, but it is being added under the overall Event Hubs API version `2023-07`, which includes other features not related to schema registry that are GA. This was the case with Json as well; the API version stayed at `2022-10` from preview to GA.

Reply:
@tg-msft, @heaths, @JeffreyRichter:  Your pespective on this would be appreciated.  My understanding was that we would want to align to a preview version.   Does the REST API board approval indicate that we're good with using a GA version here?

Reply:
I think there is a misunderstanding in regard to the preview vs GA status. From a REST api point of view the feature is considered as GA - i.e. customer using REST api to talk to schema registry to perform CRUD operations on schema group and schemas with protobuf is officially support in all regions, and changes for the feature for the REST api is final (i.e. no breaking changes). The feature is part of the GA rollout for 2023-07 so it will be available in all datacenters once the rollout is completed.

From a marketing standpoint we only labelled it as preview because the full E2E customer experience is not there yet - e.g. we still need another upgrade to support GeoDR scenario (that is not REST related), and we will get SDK support at a later date. But I don't think that influence the API versioning here?

Reply:
I'm still a bit confused.
First, Azure's api-versions use YYYY-MM-DD(-preview) format. And I only see 2023-07 here; a day should be included.

Second, I see this sentence from @nickghardwick: _This was the case with Json as well; the API version stayed at 2022-10 from preview to GA._
Azure's policy does not allow this. When going from a preview to GA, the api-version date MUST change. It is NOT OK to just remove the "-preview" suffix, the YYY-MM-DD value MUST change.

Third, Azure has versioning rules _in addition_ to breaking change rules. Whenever an API contract changes, the api-version MUST change even if the API contract changes are not breaking. I get the feeling (but I'm 100% sure) that this PR is changing the API contract (because this is what a PR does) but the api-version value is not changing when it should be.

Reply:
Thanks for the recommendation @JeffreyRichter. The YYYY-MM is a legacy versioning format pre-dated the YYYY-MM-DD format as far as I recall. I did some research in our code base, and I didn't detect any technical dependencies on this old format so we will make a change on the service to support YYYY-MM-DD (currently running through regression testing on the change, but it can go out to production as part of the upcoming deployment). @nickghardwick, please update the swagger to use 2023-07-01 instead.

We are taking the additional protobuf type as a new addition that is officially supported feature, so we are changing the version to reflect this - it is going from 2022-10 to 2023-07-01

Reply:
Latest stable release updated to '2023-07-01'

cc @hmlam, @JeffreyRichter

Reply:
Sounds good.

Reply:
@jhendrixMSFT, I see that you are assigned to this PR; is there anything else outstanding that we have to take care of before approving and merging this PR?

Reply:
The PR needs to be approved by somebody on the API stewardship board.  @JeffreyRichter is this one good to go?